clientapi/routing/register.go

```diff
@@ -310,6 +313,80 @@ func validateRecaptcha(
 	return nil
 }
 
+func validateEmailIdentity(
+	ctx context.Context,
+	cred *threepid.Credentials,
+	cfg *config.ClientAPI,
+) *util.JSONResponse {
+	if err := isTrusted(cred.IDServer, cfg); err != nil {
+		return &util.JSONResponse{
+			Code: http.StatusForbidden,
+			JSON: jsonerror.NotTrusted(cred.IDServer),
+		}
+	}
+	util.GetLogger(ctx).Infof("conecting to identity server: %s", cred.IDServer)
+	url := fmt.Sprintf(
+		"https://%s/_matrix/identity/api/v1/3pid/getValidated3pid",
```

Comment:
dendrite isn't my project, but:

this endpoint is deprecated by [MSC2713](https://github.com/matrix-org/matrix-doc/pull/2713) and it's likely that sydent will soon drop support for it.

In general, homeservers should not be delegating responsibility for email address validation to identity servers, since it allows a compromised ID server to be used to take over homeserver accounts.

Reply:
Thank you for comment. I think I have misunderstood concept of ID server. It makes sense that homeserver sends email on its own. So ID server would be user after registration to publish association, so that other may find user by email, right? I suppose also that auth stage for login should be also done without ID server. I am right?

Reply:
yes, that's right.